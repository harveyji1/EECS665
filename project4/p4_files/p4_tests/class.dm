num: int;

student: class{
    studentID: int;
};

lame: class{
    lameness: int;
};

main: () void {
    fakevariable = 1;
    num = 1;
    harvey: student;
    loser: lame;
    harvey--studentID = 1;
}