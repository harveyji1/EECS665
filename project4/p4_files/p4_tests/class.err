FATAL [2,5]: Multiply declared identifier
Name Analysis Failed
