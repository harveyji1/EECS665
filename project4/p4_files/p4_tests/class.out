num{int} : int;
student{student} : class {
    studentID{int} : int;
};
lame{lame} : class {
    lameness{int} : int;
};
main{() -> void} : () void  {
    num{int} = 1;
    harvey{student} : student{student};
    loser{lame} : lame{lame};
}
