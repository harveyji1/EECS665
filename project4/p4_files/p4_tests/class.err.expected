FATAL [12,5]-[12,17]: Undeclared identifier
Name Analysis Failed